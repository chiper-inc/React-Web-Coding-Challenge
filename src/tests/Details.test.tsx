import React from 'react';
import { shallow } from 'enzyme';
import Details from '../views/details';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

test('renders the component Home View', () => {
  const history = createMemoryHistory();
  const route = '/details/360966';
  history.push(route);
  const component = shallow(
    <Router history={history}>
      <Details />
    </Router>
  );
  expect(component).toMatchSnapshot();
});
