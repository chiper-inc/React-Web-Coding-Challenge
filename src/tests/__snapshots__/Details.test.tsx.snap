// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component Home View 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "PUSH",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "fwh0qb",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          Object {
            "hash": "",
            "key": "tdg90j",
            "pathname": "/details/360966",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 1,
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tdg90j",
          "pathname": "/details/360966",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "tdg90j",
        "pathname": "/details/360966",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "action": "PUSH",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "fwh0qb",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          Object {
            "hash": "",
            "key": "tdg90j",
            "pathname": "/details/360966",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 1,
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tdg90j",
          "pathname": "/details/360966",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Details />
  </ContextProvider>
</ContextProvider>
`;
